src/app/components/button/button.ts

```diff
@@ -81,7 +81,7 @@ export class ButtonDirective implements AfterViewInit, OnDestroy {
 
     private _internalClasses: string[] = Object.values(INTERNAL_BUTTON_CLASSES);
 
-    constructor(public el: ElementRef) {}
+    constructor(public el: ElementRef, @Inject(DOCUMENT) private document: Document) {}
```

Comment:
Hi, I don't understand how importing helps?
You must use a variable from the token, `this.document` instead of `document`(Inside the component and helpers instead of the global variable `document`)

Reply:
Hey actually it should work because then Angular SSR then knows a variable named document from the constructor if no one is present with just using document. But your absolute right that using this.document is the more correct way. Updated it :D